bitwarden_license/test/Commercial.Core.Test/AdminConsole/ProviderFeatures/RemoveOrganizationFromProviderCommandTests.cs

```diff
@@ -100,21 +103,85 @@ public class RemoveOrganizationFromProviderCommandTests
             org => org.Id == organization.Id && org.BillingEmail == "a@gmail.com"));
 
+        await stripeAdapter.Received(1).CustomerUpdateAsync(
+            organization.GatewayCustomerId, Arg.Is<CustomerUpdateOptions>(
+                options => options.Coupon == string.Empty && options.Email == "a@gmail.com"));
+
+        await sutProvider.GetDependency<IMailService>().Received(1).SendProviderUpdatePaymentMethod(
+            organization.Id,
+            organization.Name,
+            provider.Name,
+            Arg.Is<IEnumerable<string>>(emails => emails.Contains("a@gmail.com") && emails.Contains("b@gmail.com")));
+
+        await sutProvider.GetDependency<IProviderOrganizationRepository>().Received(1)
+            .DeleteAsync(providerOrganization);
+
+        await sutProvider.GetDependency<IEventService>().Received(1).LogProviderOrganizationEventAsync(
+            providerOrganization,
+            EventType.ProviderOrganization_Removed);
+    }
+
+    [Theory, BitAutoData]
+    public async Task RemoveOrganizationFromProvider_CreatesSubscriptionAndScalesSeats(Provider provider,
+        ProviderOrganization providerOrganization,
+        Organization organization,
+        SutProvider<RemoveOrganizationFromProviderCommand> sutProvider)
+    {
+        providerOrganization.ProviderId = provider.Id;
+
+        var organizationRepository = sutProvider.GetDependency<IOrganizationRepository>();
+        sutProvider.GetDependency<IOrganizationService>().HasConfirmedOwnersExceptAsync(
+                providerOrganization.OrganizationId,
+                Array.Empty<Guid>(),
+                includeProvider: false)
+            .Returns(true);
+
+        var organizationOwnerEmails = new List<string> { "a@gmail.com", "b@gmail.com" };
+
+        organizationRepository.GetOwnerEmailAddressesById(organization.Id).Returns(organizationOwnerEmails);
+
         var stripeAdapter = sutProvider.GetDependency<IStripeAdapter>();
+        stripeAdapter.SubscriptionCreateAsync(default).ReturnsForAnyArgs(new Stripe.Subscription
+        {
+            Id = "S-1",
+            CurrentPeriodEnd = DateTime.Today.AddDays(10),
+        });
 
+        await sutProvider.Sut.RemoveOrganizationFromProvider(provider, providerOrganization, organization);
         await stripeAdapter.Received(1).CustomerUpdateAsync(
             organization.GatewayCustomerId, Arg.Is<CustomerUpdateOptions>(
                 options => options.Coupon == string.Empty && options.Email == "a@gmail.com"));
 
-        await stripeAdapter.Received(1).SubscriptionUpdateAsync(
-            organization.GatewaySubscriptionId, Arg.Is<SubscriptionUpdateOptions>(
-                options => options.CollectionMethod == "send_invoice" && options.DaysUntilDue == 30));
+        var featureService = sutProvider.GetDependency<IFeatureService>();
+        var isConsolidatedBillingEnabled = featureService.IsEnabled(FeatureFlagKeys.EnableConsolidatedBilling);
```

Comment:
IIRC the feature flags are always disabled on unit tests. Please change this to validate the logic with the feature flag on or off.

Reply:
resolve